src/OrchardCore/OrchardCore.Abstractions/Shell/Builders/ShellContext.cs

```diff
@@ -48,7 +48,10 @@ public PlaceHolder()
 
         public ShellScope CreateScope()
         {
-            if (_placeHolder)
+            // We can't create a scope with a null 'ServiceProvider' meaning that the shell has been disposed. Normally, the
+            // 'ShellHost' removes the shell from its collection as soon as it is released so that a new one will be created.
+            // But this may happen when a shell releases itself its dependent shells and disposes those that are not in use.
```

Comment:
Please fix this line, do you mean:

```suggestion
            // But this may happen when a shell releases its dependent shells and disposes those that are not in use.
```